doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst

```diff
@@ -652,6 +652,87 @@ The following changes are recommended for your application to work optimally aft
 
 * For LwM2M applications, replace the :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` Kconfig option with :kconfig:option:`CONFIG_LWM2M_DTLS_CID`.
 
+* For Mesh applications, the Sensor API has been re-written, and the previous Sensor API is deprecated.
+  A new Sensor API has been created that should be used for new applications and applications that will be continuously updated.
+  The new Sensor API is described in :ref:`bt_mesh_sensors_readme`.
+  Usage is shown in :ref:`bluetooth_mesh_sensor_client`, :ref:`bluetooth_mesh_sensor_server` and :ref:`bluetooth_mesh_light_lc` samples, as they all have been updated to use the new Sensor API.
+
+  The :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE` is set default to `=y`.
+  This means for existing applications there is no need to do anything to continue using the deprecated Sensor API.
+  The application will get a deprecation warning when compiled, that the user can choose to disregard.
+  This will work until the deprecation period is over for the deprecated Sensor API.
+  When the deprecation period is over the deprecated Sensor API will be removed, and the application will no longer compile if not updated to the new Sensor API.
+
+  New and existing applications that want to use the new Sensor APIs must set :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE` to `=n` in their configuration.
+  This configuration value will be removed when the deprecation periode is over, and the configuration value then need to be removed from the application configuration.
+  It is important to check if :kconfig:option:`CONFIG_FPU` is set to `=y` to support accelerated floating point operations, and possible add :kconfig:option:`CONFIG_CBPRINTF_FP_SUPPORT` to support floating point printing.
+
+  * Application variables, Sensor API arguments and callback parameters previously defined with :c:struct:`sensor_value` now uses :c:struct:`bt_mesh_sensor_value` instead.
```

Comment:
I actual thinks the "Applicatioin variables" should be kept. The user should remember to check his/her application variables, if they should be changed to.

Reply:
Application variables typically should never be defined using `bt_mesh_sensor_value`. They should be `sensor_value`, `float`, whatever their actual use case/sensor driver provides/expects. So if you want to mention this, then you have to reword it, because "uses bt_mesh_sensor_value_instead" is definitely wrong for application variables.

Reply:
Then it is wrong in our own sample https://github.com/nrfconnect/sdk-nrf/blob/20c55d105e83c24746ed9102f846b4d9c76d3391/samples/bluetooth/mesh/sensor_server/src/model_handler.c#L63C1-L69C1

Reply:
Settings are the exception I think, but if you take a look at the lines immediately below you see that `gain` and ambient light value for example are `float` now (which was my point here). But I see your point that it should be mentioned, what about this then?

```suggestion
  * Sensor API arguments, callback parameters and relevant application variables previously defined with :c:struct:`sensor_value` now use :c:struct:`bt_mesh_sensor_value` instead.
```